execute/plugin.go

```diff
@@ -372,13 +398,34 @@ func (p *Plugin) Outcome(
 		}
 	}
 
+	decodedAos, err := decodeAttributedObservations(aos)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode observations: %w", err)
+	}
+
+	// discovery processor disabled by setting it to nil.
+	if p.discovery != nil {
+		discoveryAos := make([]plugincommon.AttributedObservation[dt.Observation], len(decodedAos))
+		for i := range decodedAos {
+			discoveryAos[i] = plugincommon.AttributedObservation[dt.Observation]{
+				OracleID:    decodedAos[i].OracleID,
+				Observation: decodedAos[i].Observation.Contracts,
+			}
+		}
+		_, err = p.discovery.Outcome(dt.Outcome{}, dt.Query{}, discoveryAos)
+		if err != nil {
+			return nil, fmt.Errorf("unable to process outcome of discovery processor: %w", err)
+		}
+		p.initialized = true
```

Comment:
Need to check that the Outcome actually did sync contracts before initializing here?

Reply:
The idea was that no error means that everything has initialized.